src/main/java/securbank/services/UserServiceImpl.java

```diff
@@ -250,4 +288,315 @@ public NewUserRequest getNewUserRequest(UUID newUserRequestId) {
 	}
+	
+	/**
+     * Creates external user modification request
+     * 
+     * @param request
+     *            The modification request to be create 
+     * @return modificationRequest
+     */
+	@Override
+	public ModificationRequest createInternalModificationRequest(User user) {
+		User current = getCurrentUser();
+		ModificationRequest request = new ModificationRequest();
+		if (user == null) {
+			logger.warn("Request for user who is not logged in");
+			
+			return null;
+		}
+		List<ModificationRequest> requests = modificationRequestDao.findAllbyUser(current);
+		logger.debug("Deactivating existing modification requests");
+		
+		// Deactivate all active request 
+		for (ModificationRequest activeRequest : requests) {
+			activeRequest.setActive(false);
+			modificationRequestDao.update(activeRequest);
+		}
+		
+		if (!user.getEmail().equals(current.getEmail())) {
+			request.setStatus("waiting");
+		}
+		else {
+			request.setStatus("pending");
+		}
+		request.setUsername(current.getUsername());
+		request.setEmail(user.getEmail());
+		request.setPhone(user.getPhone());
+		request.setFirstName(user.getFirstName());
+		request.setMiddleName(user.getMiddleName());
+		request.setLastName(user.getLastName());
+		request.setAddressLine1(user.getAddressLine1());
+		request.setAddressLine2(user.getAddressLine2());
+		request.setCity(user.getCity());
+		request.setState(user.getState());
+		request.setZip(user.getZip());
+		request.setUser(current);
+		request.setPassword(current.getPassword());
+		request.setActive(true);
+		request.setCreatedOn(LocalDateTime.now());
+		request.setUserType("internal");
+		request.setRole(user.getRole());
+		request = modificationRequestDao.save(request);
+		logger.info("Request for creating new internal user modification request");
+		
+		if (!request.getEmail().equals(current.getEmail())) {
+			message = new SimpleMailMessage(); 
+			message.setText(env.getProperty("modification.request.verify.body").replace(":id:", request.getModificationRequestId().toString()));
+			message.setSubject(env.getProperty("modification.request.verify.subject"));
+			message.setTo(request.getEmail());
+			emailService.sendEmail(message);
+		}
+		
+		return request;
+	}
+
+	/**
+     * Creates external user modification request
+     * 
+     * @param user
+     *            The modification request to be create 
+     * @return modificationRequest
+     */
+	@Override
+	public ModificationRequest createExternalModificationRequest(User user) {
+		User current = getCurrentUser();
+		if (user == null) {
+			return null;
+		}
+		ModificationRequest request = new ModificationRequest();
+		List<ModificationRequest> requests = modificationRequestDao.findAllbyUser(current);
+		
+		// Deactivate all active request 
+		for (ModificationRequest activeRequest : requests) {
+			activeRequest.setActive(false);
+			modificationRequestDao.update(activeRequest);
+		}
+		
+		if (!request.getEmail().equals(current.getEmail())) {
```

Comment:
@Nikh13 `request.getEmail()` is working for me. Email of request is set in the preceding lines. Though the we didn't have the mapping for verifying the email. 
Please confirm if this is correct.
Thanks